### ISSUE TITLE
Hairy regex cause JSON documents to be invalid.

### ISSUE THREAD

**Issue by Iheartweb** [nickcmaynard/jsonschemalint#5]
This is a valid JSON schema:
```js
{
  "type": "object",
  "properties": {
    "url": {
      "type": "string",
      "pattern": "_^(?:(?:https?|ftp)://)(?:\\S+(?::\\S*)?@)?(?:(?!10(?:\\.\\d{1,3}){3})(?!127(?:\\.\\d{1,3}){3})(?!169\\.254(?:\\.\\d{1,3}){2})(?!192\\.168(?:\\.\\d{1,3}){2})(?!172\\.(?:1[6-9]|2\\d|3[0-1])(?:\\.\\d{1,3}){2})(?:[1-9]\\d?|1\\d\\d|2[01]\\d|22[0-3])(?:\\.(?:1?\\d{1,2}|2[0-4]\\d|25[0-5])){2}(?:\\.(?:[1-9]\\d?|1\\d\\d|2[0-4]\\d|25[0-4]))|(?:(?:[a-z\\x{00a1}-\\x{ffff}0-9]+-?)*[a-z\\x{00a1}-\\x{ffff}0-9]+)(?:\\.(?:[a-z\\x{00a1}-\\x{ffff}0-9]+-?)*[a-z\\x{00a1}-\\x{ffff}0-9]+)*(?:\\.(?:[a-z\\x{00a1}-\\x{ffff}]{2,})))(?::\\d{2,5})?(?:/[^\\s]*)?$_iuS"
    }
  }
}
```

This is a JSON Document that conforms to it:
```
{
  "url": "http://www.foobar.com"
}
```

However, the above document (and all other documents) are reported as invalid JSON.

The document is reported as valid after mangling the regex like so:
```js
{
  "type": "object",
  "properties": {
    "url": {
      "type": "string",
      "pattern": "_^(?:(?:https?|ftp)://)(?:\\S+(?::\\S*)?@)"
    }
  }
}
```
But, that's not very desirable.

For info on the regex see: https://mathiasbynens.be/demo/url-regex (diegoperini)

**Comment by nickcmaynard** [comment]
JavaScript doesn't reckon your regex is a regex.

```
new RegExp("_^(?:(?:https?|ftp)://)(?:\\S+(?::\\S*)?@)?(?:(?!10(?:\\.\\d{1,3}){3})(?!127(?:\\.\\d{1,3}){3})(?!169\\.254(?:\\.\\d{1,3}){2})(?!192\\.168(?:\\.\\d{1,3}){2})(?!172\\.(?:1[6-9]|2\\d|3[0-1])(?:\\.\\d{1,3}){2})(?:[1-9]\\d?|1\\d\\d|2[01]\\d|22[0-3])(?:\\.(?:1?\\d{1,2}|2[0-4]\\d|25[0-5])){2}(?:\\.(?:[1-9]\\d?|1\\d\\d|2[0-4]\\d|25[0-4]))|(?:(?:[a-z\\x{00a1}-\\x{ffff}0-9]+-?)*[a-z\\x{00a1}-\\x{ffff}0-9]+)(?:\\.(?:[a-z\\x{00a1}-\\x{ffff}0-9]+-?)*[a-z\\x{00a1}-\\x{ffff}0-9]+)*(?:\\.(?:[a-z\\x{00a1}-\\x{ffff}]{2,})))(?::\\d{2,5})?(?:/[^\\s]*)?$_iuS")
```

**Comment by nickcmaynard** [comment]
Closing this - looks like invalid usage and no response from reporter.
